test/with_dummyserver/test_connection.py

```diff
@@ -0,0 +1,96 @@
+from typing import Generator
+
+import pytest
+
+from dummyserver.testcase import HTTPDummyServerTestCase as server
+from urllib3 import HTTPConnectionPool
+from urllib3.response import HTTPResponse
+
+
+@pytest.fixture()
+def pool() -> Generator[HTTPConnectionPool, None, None]:
+    server.setup_class()
+
+    with HTTPConnectionPool(server.host, server.port) as pool:
+        yield pool
+
+    server.teardown_class()
+
+
+def test_requires_kwargs(pool: HTTPConnectionPool) -> None:
+    conn = pool._get_conn()
+
+    with pytest.raises(TypeError) as e:
+        conn.getresponse("GET", "google.com", pool)  # type: ignore
+
+    assert "getresponse() takes 1 positional argument but 4 were given" in str(e)
+
+
+def test_returns_urllib3_HTTPResponse(pool: HTTPConnectionPool) -> None:
+    conn = pool._get_conn()
+
+    method = "GET"
+    path = "/"
+
+    conn.request(method, path)
+
+    response: HTTPResponse = conn.getresponse(
+        method=method,
+        url=path,
+        pool=pool,
+        retries=None,
+        decode_content=False,
+        enforce_content_length=False,
+        preload_content=False,
+        response_conn=None,
+    )
```

Comment:
Does this look right to you? 

I call `conn.request()` and I pass in the `method` and `url`. Then I call `conn.getresponse()` but again I have to pass in `method` and `url`? 

Does the connection not really know what response I want? Is it making the request again???

It seems to me like this could be better. Like what if:
```python
response = conn.request(method, path)
``` 
or what if returned some kinda pending response?
```python
pending_r = conn.request(method, path)
response = pending_r.getresponse()
```
___
Why does `HTTPResponse` need to know about retries? Isn't that handled by the `ConnectionPool`? Specifically by `urlopen`?
___
Why do we need so many options for reading the response? It seems to me would replace all these with somekinda config object.

```python
r_config = HTTPResponseConfig() This would have same defaults we use today

conn.getresponse(r_config)
```

You could then customise `HTTPResponseConfig` in the constructor or the java way lol
```python
response = r_config.with_conn(conn).with_decode().with_enforce_content_length().getresponse()
```
___

NONE of the above is for this PR <3, just food for thought on potential future enhancements.